### ISSUE TITLE
What warrants an incident with coveralls?

### ISSUE THREAD

**Issue by oshi97** [lemurheavy/coveralls-public#1643]
Yesterday, our team noticed that Coveralls was experiencing very bad latency with regards to PR response time.
This was across all our repos, not just a single one.
example where the coveralls check took nearly 3 hours: https://github.com/yext/answers-react-components/pull/173

I also see that today Coveralls was sitting at over 4000 seconds for coverage calculation job times, which is over an hour.
![image](https://user-images.githubusercontent.com/23005393/173676415-48204801-be9f-4482-b699-f3bdad2d3add.png)

We're not paying for Coveralls at all except for arguably free promotion, but wouldn't a situation like this warrant some kind of incident on the status page? Or is this well within operating standards.

**Comment by afinetooth** [comment]
Hi, @oshi97. 

Yes, these numbers definitely constitute what should have been an incident. We should have received a notification once we had 2-3 instances around those peaks. I'm not sure why we didn't. But I was on yesterday, so it's possible I missed something. 

Let me do a quick postmortem on this and see if I can figure out what happened.

In the meantime, I can see that we're at a normal traffic level but a rather long average latency for that traffic level, so I'll try to address current performance issues first.

If you get this today, are you experiencing the same sort of latency today?

Thanks for bringing this to our attention. In the future, paying customer or not, if your team sees this kind of latency on pull request comments more than 2-3 times in a row, especially if across multiple projects, I'd encourage you to shoot us an email at support@coveralls.io. You'll be doing us a favor and we'll respond pretty quickly with protocol if we see anything like, "performance", "latency" or "slow PR comments" in the subject.

**Comment by oshi97** [comment]
Thanks so much! It's nice to know this is important for you guys. No latency today everything is speedy 👍

**Comment by afinetooth** [comment]
**Quick update / postmortem**:
The nature of the performance issues we had weren't related to traffic, directly, and therefore weren't captured by the the triggers we use to auto-scale. The dequeue times definitely slowed, but those aren't the full picture. Instead, the issue appears to be with our database layer. Briefly, we have an enormous database that more frequently than you'd think, has tables grow beyond rates that can be easily maintained with the regular auto-vacuum procedure. 

We posted an incident right after your report on **Wed (Jun 15)**, here:
https://status.coveralls.io/incidents/krrc78mthkyb

Then, after that, started a couple of manual maintenance procedures, which helped, but we experienced the issue again this morning, and so just published another incident **today (Jun 17)**:
https://status.coveralls.io/incidents/bh491q26c220

We are continuing with some manual maintenance procedures, which will help but are themselves occupy the database in a way that exacerbates the issue, so we've planned more thorough maintenance for **this Sat (Jun 18) from approx. 12pm-4pm US PST**. If all goes as planned, there should be no downtime, but we've planned for a worst-case scenario that we think would mean ~30-min downtime, during which coveralls.io would be available in _read-only_ mode.

We'll post, and update, about that on Sat, so, particularly if you're a weekend user, now is a good time to check out and subscribe to our status page:
https://status.coveralls.io/
